pkg/operator/apis/monitoring/v1alpha1/types_test.go

```diff
@@ -124,7 +128,99 @@ func TestValidatePodMonitoring(t *testing.T) {
 				},
 			},
 			fail:        true,
-			errContains: "conflicts with GMP target schema",
+			errContains: `invalid PodMonitoring target labels: cannot relabel with action "replace" onto protected label "cluster"`,
+		}, {
+			// A simple error that should be caught by invoking the upstream validation. We don't
+			// have to cover everything it covers.
+			desc: "remapping onto bad label name",
+			pm: PodMonitoringSpec{
+				Endpoints: []ScrapeEndpoint{
+					{
+						Port:     intstr.FromString("web"),
+						Interval: "10s",
+					},
+				},
+				TargetLabels: TargetLabels{
+					FromPod: []LabelMapping{
+						{From: "key1", To: "foo-bar"},
+					},
+				},
+			},
+			fail:        true,
+			errContains: `"foo-bar" is invalid 'target_label' for replace action`,
```

Comment:
IIUC this error is formatted and returned by the Prometheus library rather than prometheus-engine code - just wanted to confirm.

Reply:
Yes, so this is in principle at risk of breaking with dependency updates. But that'll be easy to fix and I the error validation in this test is necessary to confirm we triggered the intended logic and don't get a false positive due to a different error.